### ISSUE TITLE
Bash completion

### ISSUE THREAD

**Issue by braderhart** [apache/cloudstack-cloudmonkey#62]
I noticed today when trying this out on Windows, that the interactive console is slow. I tried pasting some text and I could watch the characters being typed. It took about 4 seconds to paste 25 characters. It works descent in Linux, but still produces artifacts when tab completing. Would love to see bash completion scripts as to no rely on the interactive console (e.g. Docker, Kubernetes, Helm, etc).

**Comment by braderhart** [comment]
BTW I created a package for Arch:

https://aur.archlinux.org/packages/cloudmonkey-bin/

**Comment by rhtyd** [comment]
@braderhart does it make any difference if you use cmk from the powershell terminal?

**Comment by onitake** [comment]
It's a bit unfortunate that cloudmonkey doesn't use one of the standard Go flag packages (such as https://github.com/spf13/cobra ). This would make it much easier to add bash completion support.

On the other hand, it shouldn't be too difficult to use something like https://github.com/posener/complete directly.

**Comment by braderhart** [comment]
Yeah agreed. Hugo, kubectl, eksctl, skaffold, and others work great.

**Comment by rhtyd** [comment]
@onitake cobra is not a standard Go flag package, besides it's a heavily bloated dependency.

`cmk` provides an autocompletion for not just parameters but also parameter values when used in the shell mode. I'm not sure other stated dependencies provide value completion (by querying via a server-side request) as well. Pull requests are welcome!

**Comment by onitake** [comment]
@rhtyd I totally agree with you on that one. However, by using cobra, bash autocompletion would be easier to add.
The standard flag package is very primitive and cannot autogenerate bash autocompletion strings like many Go tools do.

The semi-dynamic nature of cmk will certainly make it a bit more difficult to implement this - but there are many tools that do send server requests when autocompleting. `scp` and `kubectl` come to mind.

`posener/complete` is compatible with the standard flags package, so integrating it into `cmk` will be easy. What will be a bit more complicated is adding the dynamic commands. The documentation is here: https://godoc.org/github.com/posener/complete
